pkg/test/e2e/ccm-migration/ccm_migration_test.go

```diff
@@ -40,117 +40,156 @@ import (
-				if err := clusterJig.SeedClient.List(context.Background(), machineControllerWebhookPods, ctrlruntimeclient.InNamespace(clusterJig.Cluster.Status.NamespaceName), ctrlruntimeclient.MatchingLabels{
-					resources.AppLabelKey: resources.MachineControllerWebhookDeploymentName,
-				}); err != nil {
-					return false, err
-				}
-				if len(machineControllerWebhookPods.Items) != 1 {
-					return false, nil
-				}
-				for _, arg := range machineControllerWebhookPods.Items[0].Spec.Containers[0].Args {
-					if strings.Contains(arg, "-node-external-cloud-provider") {
-						return true, nil
-					}
-				}
-				return false, nil
-			})).NotTo(gomega.HaveOccurred())
-
-			machines := &clusterv1alpha1.MachineList{}
-			ginkgo.By("rolling out all the machines")
-			gomega.Expect(userClient.List(context.Background(), machines)).NotTo(gomega.HaveOccurred())
-			for _, m := range machines.Items {
-				gomega.Expect(userClient.Delete(context.Background(), &m)).NotTo(gomega.HaveOccurred())
+			testBody(clusterJig, vsphereCluster, userClient)
+		})
+	})
+
+	ginkgo.Context("OpenStack provider", func() {
+		var (
+			openstackCluster *kubermaticv1.Cluster
+		)
+
+		ginkgo.BeforeEach(func() {
+			if options.provider != string(kubermaticv1.OpenstackCloudProvider) {
+				ginkgo.Skip("skipping vsphere test")
 			}
+			openstackCluster = &kubermaticv1.Cluster{}
+			clusterJig = providers.NewClusterJigOpenstack(seedClient, options.kubernetesVersion, options.osSeedDatacenter, options.osCredentials)
+			userClient = setupAndGetUserClient(clusterJig, openstackCluster, clusterClientProvider)
+		})
 
-			ginkgo.By("waiting for the complete cluster migration")
-			gomega.Expect(wait.Poll(userClusterPollInterval, customTestTimeout, func() (done bool, err error) {
-				migratingCluster := &kubermaticv1.Cluster{}
-				if err := clusterJig.SeedClient.Get(context.Background(), types.NamespacedName{Name: clusterJig.Cluster.Name}, migratingCluster); err != nil {
-					return false, err
-				}
+		ginkgo.AfterEach(func() {
+			gomega.Expect(clusterJig.Cleanup(userClient)).NotTo(gomega.HaveOccurred())
+		})
 
-				return migratingCluster.Status.Conditions[kubermaticv1.ClusterConditionCSIKubeletMigrationCompleted].Status == corev1.ConditionTrue, nil
-			})).NotTo(gomega.HaveOccurred())
+		ginkgo.It("migrating cluster to external CCM", func() {
+			testBody(clusterJig, openstackCluster, userClient)
 		})
 	})
 })
+
+func setupAndGetUserClient(clusterJig providers.ClusterJigInterface, cluster *kubermaticv1.Cluster, clusterClientProvider *clusterclient.Provider) ctrlruntimeclient.Client {
+	ginkgo.By("setting up cluster")
+	gomega.Expect(clusterJig.Setup()).NotTo(gomega.HaveOccurred(), "user cluster should deploy successfully")
+	clusterJig.Log().Debugw("Cluster set up", "name", clusterJig.Name())
+	gomega.Expect(clusterJig.Seed().Get(context.TODO(), types.NamespacedName{Name: clusterJig.Name()}, cluster)).NotTo(gomega.HaveOccurred())
```

Comment:
Please switch to `context.Background()` and put it in some variable, so you can reuse it.